### PR TITLE
Add perun import command

### PR DESCRIPTION
**Pull request by tfiedor**

### CODE REVIEW

File: perun/cli_groups/import_cli.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+from perun.profile import imports
+from perun.utils.common import cli_kit
+
+
+@click.group("import")
+@click.option(
+    "--machine-info",
+    "-m",
+    type=click.Path(resolve_path=True, readable=True),
+    help="Imports machine info from file in JSON format (by default, machine info is loaded from the current hoot)."
```
Comment (JiriPavela) [inline perun/cli_groups/import_cli.py]: Seems like there is a typo: hoot -> host?

File: perun/cli_groups/import_cli.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    help="Sets [INT] samplings of the ith profiled command.",
+)
+@click.pass_context
+def perf_group(ctx: click.Context, **kwargs: Any) -> None:
+    """Imports Perun profiles from perf results
+
+    This supports either profiles collected in:
+
+      1. Binary format: e.g., `collected.data` files, that are results of `perf record`
+      2. Text format: result of `perf script` that parses the binary into user-friendly and parsing-frinedly text format
```
Comment (JiriPavela) [inline perun/cli_groups/import_cli.py]: Another typo: frinedly -> friendly.

File: perun/profile/imports.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    """
+    if machine_info is not None:
+        with open(machine_info, "r") as machine_handle:
+            return json.load(machine_handle)
+    else:
+        return environment.get_machine_specification()
+
+
+def get_param(cfg: dict[str, Any], param: str, index: int) -> Any:
+    """
```
Comment (JiriPavela) [inline perun/profile/imports.py]: The docstring is missing a short description. The function seems rather simple and understandable on its own, but still, should we add at least a short description here?

File: utils/generate_machine_info.sh [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  echo -n -e "]"
+
+  echo -n "}"
+}
+
+if [[ -z "$output_file" ]]; then
+  get_machine_specification | sed 's/\\/\\\\/g'
+else
+  get_machine_specification | sed 's/\\/\\\\/g' > "$output_file"
+fi
```
Comment (JiriPavela) [inline utils/generate_machine_info.sh]: Seems like we're missing a newline here. Pre-commit is set up such that it checks only `*.py` files so it doesn't accidentally change something it shouldn't. Maybe we could add `*.sh` files as well?

**Review by JiriPavela — APPROVED** [review]
Great, adding support for importing external profiles seems like a good idea.

Maybe a quick idea for a slight improvement: would it make sense to allow the user to import the profiling configuration (i.e., cmd, workload, warmups, ...) using a single file, similarly to how the machine info is imported? I can see it being a bit mundane having to specify numerous parameters in the future if / when we decide to support more external profiles, e.g., Valgrind.

File: utils/generate_machine_info.sh (hunk 4 above)
Comment (tfiedor) [reply utils/generate_machine_info.sh]: Sure, I will let you add this to precommit hook  in separate PR. I will fix the missing new line though.

**Comment by tfiedor** [comment]
> Great, adding support for importing external profiles seems like a good idea.
> 
> Maybe a quick idea for a slight improvement: would it make sense to allow the user to import the profiling configuration (i.e., cmd, workload, warmups, ...) using a single file, similarly to how the machine info is imported? I can see it being a bit mundane having to specify numerous parameters in the future if / when we decide to support more external profiles, e.g., Valgrind.

This is good idea, but I would probably make this more generic and solve it in different PR. For now it is like this after consultation with JH, who was not a fan of the additional file :)

**Pull request merged**